plugins/event_source/sqs.py

```diff
@@ -0,0 +1,74 @@
+"""
+sqs.py
+
+An ansible-rulebook event source plugin for receiving events via an AWS SQS queue.
+
+Arguments:
+    region: AWS region containing the SQS queue
+    queue:  Name of the SQS queue
+"""
+
+import asyncio
+import logging
+import sys
+
+from typing import Any, Dict
+
+import botocore.exceptions
+
+from aiobotocore.session import get_session
+
+
+async def main(queue: asyncio.Queue, args: Dict[str, Any]):
+    logger = logging.getLogger()
+
+    region = args.get("region")
+    queue_name = args.get("queue")
+
+    # Boto should get credentials from ~/.aws/credentials or the environment
+    session = get_session()
+    async with session.create_client('sqs', region_name=region) as client:
+        try:
+            response = await client.get_queue_url(QueueName=queue_name)
+        except botocore.exceptions.ClientError as err:
+            if (
+                err.response['Error']['Code']
+                == 'AWS.SimpleQueueService.NonExistentQueue'
+            ):
+                logger.critical("Queue %s does not exist", queue_name)
+                sys.exit(1)
```

Comment:
Do not exit here. This is a plugin. It is safe to raise an exception in plugin. Rulebook engine will detect the exception and exit properly.